docs/source/guides/native_assets_sdk.md

```diff
@@ -0,0 +1,411 @@
+                .setPolicyId(policyId)
+                .build();
+
+        CompletableFuture<NativeAssetTransaction> completableFuture = new CompletableFuture();
+        api.transferAsset(nativeAssetId, toAddress, amount, depth, lastResult(completableFuture,
+                r -> r.getAssetTx()));
+        return completableFuture;
+
+    }
+```
+**note** check observer sample for `lastResult`
+
+
+### Fund the address associated with the policy
+* Replace `policyId` with the `policyId`
+* Replace `amount` with amount of maximum lovelace to fund 
+* Replace `depth` with the number of network confirmations required to treat the transaction as Confirmed
+```java
+    public CompletionStage<FundNativeAssetTransaction> fund() {
+        String policyId = "policyId";
+        Long amount = 10l;
+        Integer depth = 4;
+
+        CompletableFuture<FundNativeAssetTransaction> completableFuture = new CompletableFuture();
+        api.fundPolicyAddress(policyId, amount, depth, lastResult(completableFuture,
+                r -> r.getFundTx()));
+        return completableFuture;
+
+    }
+```
+**note** check observer sample for `lastResult`
+
+### Create an Airdrop
+* Replace `policyId` with the `policyId`
+* Replace `assetName` with arbitrary `token name`
+* Replace `targetAddress` with the address the assets will be sent to
+* Replace `amount` with amount of asset to airdrop
+```java
+    public CompletionStage<String> sendAirDrop() {
+        String policyId = "policyId";
+        String assetName = "assetName";
+        String targetAddress = "targetAddress";
+        Long amount = 10l;
+
+        List<AirDrop> airDrops = List.of(AirDrop.newBuilder()
+                .setPolicyId(policyId)
+                .setAssetName(assetName)
+                .setAddress(targetAddress)
+                .setAmountToAirDrop(amount)
+                .build());
+        
+        //Some Metadata
+        Map<String, Value> data = new HashMap<>();
+        Struct struct = com.google.protobuf.Struct.newBuilder()
+                .putAllFields(data)
+                .build();
+
+        return api.sendAirDropBatch(struct, airDrops)
+                .thenApply(response -> {
+                    if (response.getProblem().getMsg().isEmpty())
+                        return response.getBatchId();
+                    else {
+                        throw new RuntimeException(response.getProblem().getMsg());
+                    }
+                });
+
+    }
+```
+
+### Check the airdrop status
+* Replace `batchId` with the `batchId` of Airdrop
+```java
+    public CompletionStage<AirDropStatusResponse> airDropStatus() {
+        String batchId = "batchId";
+        return api.getAirDropStatus(batchId);
+
+    }
+```
+
+### Observer sample used in snippets above
+```java
+public <E, R> StreamObserver<E> lastResult(CompletableFuture<R> result, Function<E, R> f) {
+        return new StreamObserver<>() {
+            ArrayList<R> buff = new ArrayList();
```

Comment:
Since we care only about last result:
```suggestion
            R lastResult;
```
After this change onNext() and onCompleted() need to be changed accordingly